display_list/display_list_image_filter.h

```diff
@@ -617,6 +654,79 @@ class DlColorFilterImageFilter final : public DlImageFilter {
   std::shared_ptr<DlColorFilter> color_filter_;
 };
 
+class DlLocalMatrixImageFilter final : public DlImageFilter {
+ public:
+  explicit DlLocalMatrixImageFilter(const SkMatrix& matrix,
+                                    std::shared_ptr<DlImageFilter> filter)
+      : matrix_(matrix), image_filter_(filter) {}
+  explicit DlLocalMatrixImageFilter(const DlLocalMatrixImageFilter* filter)
+      : DlLocalMatrixImageFilter(filter->matrix_, filter->image_filter_) {}
+  DlLocalMatrixImageFilter(const DlLocalMatrixImageFilter& filter)
+      : DlLocalMatrixImageFilter(&filter) {}
+  std::shared_ptr<DlImageFilter> shared() const override {
+    return std::make_shared<DlLocalMatrixImageFilter>(this);
+  }
+
+  DlImageFilterType type() const override {
+    return DlImageFilterType::kLocalMatrixFilter;
+  }
+  size_t size() const override { return sizeof(*this); }
+
+  const SkMatrix& matrix() const { return matrix_; }
+
+  const DlLocalMatrixImageFilter* asLocalMatrix() const override {
+    return this;
+  }
+
+  bool modifies_transparent_black() const override { return false; }
+
+  SkRect* map_local_bounds(const SkRect& input_bounds,
+                           SkRect& output_bounds) const override {
+    if (!image_filter_) {
+      return nullptr;
+    }
+    return image_filter_->map_local_bounds(input_bounds, output_bounds);
+  }
+
+  SkIRect* map_device_bounds(const SkIRect& input_bounds,
+                             const SkMatrix& ctm,
+                             SkIRect& output_bounds) const override {
+    if (!image_filter_) {
+      return nullptr;
+    }
+    return image_filter_->map_device_bounds(
+        input_bounds, SkMatrix::Concat(ctm, matrix_), output_bounds);
+  }
+
+  SkIRect* get_input_device_bounds(const SkIRect& output_bounds,
+                                   const SkMatrix& ctm,
+                                   SkIRect& input_bounds) const override {
+    if (!image_filter_) {
+      return nullptr;
+    }
+    return image_filter_->get_input_device_bounds(
+        output_bounds, SkMatrix::Concat(ctm, matrix_), input_bounds);
+  }
+
+  sk_sp<SkImageFilter> skia_object() const override {
+    if (!image_filter_) {
+      return nullptr;
+    }
+    return image_filter_->skia_object()->makeWithLocalMatrix(matrix_);
+  }
+
+ protected:
+  bool equals_(const DlImageFilter& other) const override {
+    FML_DCHECK(other.type() == DlImageFilterType::kMatrix);
+    auto that = static_cast<const DlLocalMatrixImageFilter*>(&other);
+    return (matrix_ == that->matrix_ && image_filter_ == that->image_filter_);
```

Comment:
`*image_filter_ == *(that->image_filter_)`, or use the Equals helper mechanism.

Reply:
You will need null checks as well, but I think Equals does that for you returning false if only one of them is null.